acceptance/suites/tests/555-puppetserver-service/reboot.rb

```diff
@@ -0,0 +1,55 @@
+if (options[:type] == 'pe') then
+  pssn='pe-puppetserver'
+else
+  pssn='puppetserver'
+end
+
+test_name "Validates the ability of the #{pssn} service to reliably start after a reboot."
+
+max=3
+sleeptime=45
+retryCounter=0
+step "\n-----------\nValidate #{pssn} is up and running, then reboot, repeat #{max} times\n-----------"
+
+for i in 1..max 
+  step "\n-----\nLoop #{i}\n-----"
+   
+  step "check uptime"
+  origUptime=on(master, "awk '{print $1}' /proc/uptime").stdout.chomp.to_i
+
+  step "reboot"
+  #rebooting with on(master,'reboot') causes an IOError: stream end.  This seems like a sane workaround.
+  #QENG-1914 for more...
+  system "ssh -i #{options[:keyfile]} -o StrictHostKeyChecking=no root@#{master.reachable_name} \'reboot\'"
+
+  step "sleeping for 15 seconds and waiting for reboot."
+  sleep 15
+
+  unless port_open_within?(master,22,120)
+    raise Beaker::DSL::FailTest, 'Could not SSH into puppet master within 2 minutes after reboot'
```

Comment:
This doens't actually try and SSH in, correct?  If it only tests to make sure the port is open then the message should reflect that.

This is important because there exists a window in time where the socket is open and listening, but the SSH server is not up and running yet so attempting to SSH in will fail while testing if the socket is open will succeed.